src/fmeval/eval_algorithms/summarization_accuracy.py

```diff
@@ -178,41 +168,34 @@ def evaluate(
 
         :return: List of EvalOutput objects.
         """
-        if dataset_config:
-            dataset_configs = [dataset_config]
-        else:
-            dataset_configs = [DATASET_CONFIGS[dataset_name] for dataset_name in EVAL_DATASETS[self.eval_name]]
-
+        dataset_configs = get_dataset_configs(dataset_config, self.eval_name)
```

Comment:
I'm not following here. what if `dataset_config` has been passed as argument?
Also. I think it would be much cleaner to have under the hood a generic implementation, with dataset taken as required argument and a built-in version that calls the generic version on the built-in datasets. Is this possible?